### DISCUSSION TITLE
unable to successfully compile stable release 4.2.1 ???

### DISCUSSION THREAD

**Discussion by MichaelWeaser** [86Box/86Box#5045] [General]
Technically it compiles successfully but it will not link??? 

seems to be an issue with libopenal???

It errors out with this : 

$ cmake --build ~/86Box/build
[1/1] Linking CXX executable src\86Box.exe
FAILED: src/86Box.exe
C:\WINDOWS\system32\cmd.exe /C "cd . && C:\msys64\mingw64\bin\c++.exe -O3 -DNDEBUG -mwindows  -static -Wl,--allow-multiple-definition @CMakeFiles\86Box.rsp -o src\86Box.exe -Wl,--out-implib,src\lib86Box.dll.a -Wl,--major-image-version,0,--minor-image-version,0 && cd ."
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/14.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:/msys64/mingw64/lib/libopenal.a(wasapi.cpp.obj):(.text+0x54e3): undefined reference to `AvSetMmThreadCharacteristicsW'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/14.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:/msys64/mingw64/lib/libopenal.a(wasapi.cpp.obj):(.text+0x5620): undefined reference to `AvRevertMmThreadCharacteristics'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/14.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:/msys64/mingw64/lib/libopenal.a(wasapi.cpp.obj):(.text+0x5a98): undefined reference to `AvRevertMmThreadCharacteristics'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/14.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:/msys64/mingw64/lib/libopenal.a(wasapi.cpp.obj):(.text+0xa792): undefined reference to `AvSetMmThreadCharacteristicsW'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/14.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:/msys64/mingw64/lib/libopenal.a(wasapi.cpp.obj):(.text+0xa975): undefined reference to `AvRevertMmThreadCharacteristics'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/14.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:/msys64/mingw64/lib/libopenal.a(wasapi.cpp.obj):(.text+0xb28d): undefined reference to `AvRevertMmThreadCharacteristics'
collect2.exe: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.

**Comment by jriwanek** [comment]
4.2.1 needs an older version of openAL, just use the latest code from the github.

**Reply by MichaelWeaser** [reply]
What older version of openAL is needed??? Says something about that openAl is optional???

**Reply by lemondrops** [reply]
1.23.1, but you could just backport https://github.com/86Box/86Box/commit/8f68703255047b7fa2bacff9e5f9299c047f6795, if you insist on building the stable version instead of the latest code.
